internal/core/src/common/ChunkWriter.cpp

```diff
@@ -176,7 +176,7 @@ ArrayChunkWriter::write(std::shared_ptr<arrow::RecordBatchReader> data) {
             arrays.push_back(std::move(arr));
             if (is_string) {
                 // element offsets size
-                size += sizeof(uint64_t) * arr.length();
+                size += sizeof(uint32_t) * arr.length();
             }
```

Comment:
only change inner elements's offsets from uint64 to uint32
but I suppose the offsets within one chunk cannot exceed uint32 also